internal/common/objects/Audi.go

```diff
@@ -0,0 +1,72 @@
+package objects
+
+import (
+	phy "SOMAS2023/internal/common/physics"
+	"SOMAS2023/internal/common/utils"
+	"math"
+)
+
+type IAudi interface {
+	IBaseBiker
+	IPhysicsObject
+}
+
+type Audi struct {
+	*BaseBiker
+	*PhysicsObject
+	target *MegaBike
+}
+
+// GetAudi is a constructor for Audi that initializes it with a new UUID and default position.
+func GetAudi() *Audi {
+	return &Audi{
+		PhysicsObject: GetPhysicsObject(utils.MassAudi),
+	}
+}
+
+func GetIAudi() IAudi {
+	return &Audi{
+		PhysicsObject: GetPhysicsObject(utils.MassAudi),
+	}
+}
+
+// Move as MegaBike, called by server
+func (audi *Audi) Move() {
+	audi.coordinates = phy.GetNewPosition(audi.coordinates, audi.velocity, audi.orientation)
+}
+
+// DecideForce is called by server as an agent, Audi finds possible target and move towards it at constant velocity
```

Comment:
Since Audi is not an agent it doesn't need to DecideForce. The plan is that the server will simply call: MegaBike.move() and Audi.move(). So I would suggest you move this to the Move function - like we did for the MegaBike.